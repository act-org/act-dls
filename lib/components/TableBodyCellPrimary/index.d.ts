/**
 * @prettier
 */
import * as React from 'react';
import { TableCellProps } from '@material-ui/core';
import { SortObject } from '~/types';
export interface Props<T> extends TableCellProps {
    dataKey?: string;
    sortObject?: SortObject<T>;
}
declare const TableBodyCellPrimary: <T>({ dataKey, sortObject, ...otherProps }: Props<T>) => React.ReactElement<any>;
export default TableBodyCellPrimary;
//# sourceMappingURL=index.d.ts.map