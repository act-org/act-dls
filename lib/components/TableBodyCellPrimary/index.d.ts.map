{"version":3,"file":"index.d.ts","sourceRoot":"","sources":["../../../src/components/TableBodyCellPrimary/index.tsx"],"names":[],"mappings":"AAAA;;GAEG;AAEH,OAAO,KAAK,KAAK,MAAM,OAAO,CAAC;AAE/B,OAAO,EAAa,cAAc,EAAE,MAAM,mBAAmB,CAAC;AAE9D,OAAO,EAAE,UAAU,EAAE,MAAM,SAAS,CAAC;AAIrC,MAAM,WAAW,KAAK,CAAC,CAAC,CAAE,SAAQ,cAAc;IAC9C,OAAO,CAAC,EAAE,MAAM,CAAC;IACjB,UAAU,CAAC,EAAE,UAAU,CAAC,CAAC,CAAC,CAAC;CAC5B;AAED,QAAA,MAAM,oBAAoB,2DAIZ,KAAK,CAAC,YAAY,CAAC,GAAG,CAoBnC,CAAC;AAEF,eAAe,oBAAoB,CAAC"}