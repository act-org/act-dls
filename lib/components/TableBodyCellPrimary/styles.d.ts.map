{"version":3,"file":"styles.d.ts","sourceRoot":"","sources":["../../../src/components/TableBodyCellPrimary/styles.ts"],"names":[],"mappings":"AAAA;;GAEG;;AAOH,wBASI"}