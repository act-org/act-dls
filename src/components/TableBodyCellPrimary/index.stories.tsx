/**
 * @prettier
 */

/* eslint-disable sort-keys */

import * as React from 'react';
import { select, text } from '@storybook/addon-knobs';

import TableBodyCellPrimary from '.';

export const primary = (): React.ReactElement<any> => (
  <TableBodyCellPrimary
    dataKey="itemKey"
    sortObject={{
      dataKey: 'itemKey',
      direction: select(
        'Sort Direction',
        {
          Ascending: 'ASC',
          Descending: 'DESC',
        },
        'ASC',
      ),
    }}
  >
    {text('Text', 'Cell')}
  </TableBodyCellPrimary>
);

export default {
  component: TableBodyCellPrimary,
  parameters: {
    info: {
      text: 'This is the primary variant of the table body cell component.',
    },
  },
  title: 'Atoms/TableBodyCell',
};
