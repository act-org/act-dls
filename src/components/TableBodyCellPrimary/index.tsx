/**
 * @prettier
 */

import * as React from 'react';
import clsx from 'clsx';
import { TableCell, TableCellProps } from '@material-ui/core';

import { SortObject } from '~/types';

import useStyles from './styles';

export interface Props<T> extends TableCellProps {
  dataKey?: string;
  sortObject?: SortObject<T>;
}

const TableBodyCellPrimary = <T,>({
  dataKey,
  sortObject,
  ...otherProps
}: Props<T>): React.ReactElement<any> => {
  const sortIsApplied: boolean =
    sortObject && dataKey ? sortObject.dataKey === dataKey : false;

  const classes = useStyles();

  return (
    <TableCell
      classes={{
        root: clsx(
          classes.tableCellRoot,
          sortIsApplied && classes.tableCellRootSortApplied,
        ),
      }}
      component="td"
      padding="none"
      scope="row"
      {...otherProps}
    />
  );
};

export default TableBodyCellPrimary;
