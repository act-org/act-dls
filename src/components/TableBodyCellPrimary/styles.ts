/**
 * @prettier
 */

import { grey } from '@material-ui/core/colors';
import { makeStyles } from '@material-ui/core/styles';

import DIMS from '~/constants/dims';

export default makeStyles(({ typography }: any) => ({
  tableCellRoot: {
    fontSize: '0.888rem',
    fontWeight: typography.fontWeightRegular,
    padding: `${DIMS.LAYOUT_PADDING}px !important`,
  },
  tableCellRootSortApplied: {
    backgroundColor: grey[100],
  },
}));
