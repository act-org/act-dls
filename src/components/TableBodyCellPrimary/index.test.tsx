/**
 * @prettier
 */

import * as React from 'react';
import renderer from 'react-test-renderer';

import TableBodyCellPrimary from '.';

describe('TableBodyCellPrimary', () => {
  it('matches the snapshot', () => {
    const tree = renderer
      .create(
        <TableBodyCellPrimary
          dataKey="itemKey"
          sortObject={{
            dataKey: 'itemKey',
            direction: 'ASC',
          }}
        >
          child
        </TableBodyCellPrimary>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
