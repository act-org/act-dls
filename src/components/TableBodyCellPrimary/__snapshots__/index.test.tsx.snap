// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableBodyCell matches the snapshot 1`] = `
<td
  aria-sort={null}
  className="MuiTableCell-root makeStyles-tableCellRoot-1 makeStyles-tableCellRootSortApplied-2 MuiTableCell-paddingNone"
  role="cell"
  scope="row"
>
  child
</td>
`;
